Kernel/Syscalls/pipe.cpp

```diff
@@ -19,7 +19,7 @@ KResultOr<FlatPtr> Process::sys$pipe(int pipefd[2], int flags)
     if ((flags & (O_CLOEXEC | O_NONBLOCK)) != flags)
         return EINVAL;
 
-    u32 fd_flags = (flags & O_CLOEXEC) ? FD_CLOEXEC : 0;
+    u32 fd_flags = flags & (O_CLOEXEC | O_NONBLOCK);
```

Comment:
Note that the flag being set here is `FD_CLOEXEC`, _not_ `O_CLOEXEC`. I doubt that this works properly.

Reply:
This is one of my least favorite parts of POSIX: file flags vs file description flags.

One set is retrieved via fcntl(GET_FL) and the other set is retrieved via fcntl(GET_FD) and I never remember which is which. I think that O_CLOEXEC is one of the GET_FD flags, while O_NONBLOCK is one of the GET_FL flags?

Reply:
That's correct; `O_NONBLOCK` is a file status flag, while `O_CLOEXEC` is a file descriptor flag. I understand the reasoning for the distinction but the names are indeed very easy to confuse.

Reply:
I see... I am closing this since from my reading of fcntl man page, only FD_CLOEXEC is valid here for setting fd flags? So the current code is correct I believe. Thanks for the clarification.

Reply:
I think the best way to figure out if the original code or your changed code is correct would be to write a test using `fcntl` 😋